.github/workflows/doc.yml

```diff
@@ -37,3 +39,47 @@ jobs:
           git diff-index --quiet HEAD || git commit -m "Updates pyoxigraph documentation"
           git push
         working-directory: ./website
+
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1 
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{github.actor}}
+          password: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Generate image ID and version
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/oxigraph
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "master" ] && VERSION=latest
+          echo "IMAGE_ID=$IMAGE_ID" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
```

Comment:
Is it really useful to have a big bash script for that? It might be more convenient to just use `ghcr.io/oxigraph/oxigraph:latest` in the `docker/build-push-action` action call.

Reply:
> The only thing that I am not sure and didn't really think of, is that in the end you will have 2 different images on 2 different repository: release on DockerHub and nightly on GitHub Container Registry, which could be slightly confusing (ideally you should just have 1 repository with 1 latest tag and all the release tags)

An option: have on DockerHub the release (`vX.Y.Z` tags and `latest` tag for the latest release) and in GitHub Container Registry the release (`vX.Y.Z` tags and `latest` tag for the latest release) and the nightly builds (`nightly` tag).
Using the `latest` tag only for release would make the users pull by default the latest stable release, it seems a bit safer to me than pulling the lastest nightly build.

Reply:
It makes perfect sense @Tpt , the user should get a stable release when pulling without providing a specific tag

I moved the gh release publish to `release.yml` (I re-used the step you had for the dockerhub build)

And I set the `doc.yml` to publish to the nightly tag, note that by default the [docker-metadata action](https://github.com/docker/metadata-action#basic) would use the `master` tag, that might be more semantically accurate, but I am not sure what is the usual convention for this

I currently enabled the github action caching  for the `nightly` tag action, it can be quite good to reduce regular builds time, but it's still experimental, so you might want to disable it if you are facing issues. I added a comment with a link to the doc about it in the action file